bee-network/bee-autopeering/src/init.rs

```diff
@@ -0,0 +1,203 @@
+    log::info!("Current time: {}", time::datetime_now());
+    log::info!("Private salt: {}", private_salt);
+    log::info!("Public salt: {}", public_salt);
+    log::info!("Bind address: {}", config.bind_addr());
+
+    // Create or load a peer store.
+    let peerstore = S::new(peerstore_config);
+
+    // Create peer lists.
+    let master_peers = MasterPeersList::default();
+    let active_peers = ActivePeersList::default();
+    let replacements = ReplacementList::default();
+
+    // Create a task manager to have good control over the tokio task spawning business.
+    let mut task_mngr = TaskManager::<_, NUM_TASKS>::new(peerstore.clone(), active_peers.clone(), replacements.clone());
+
+    // Create channels for inbound/outbound communication with the UDP server.
+    let (discovery_tx, discovery_rx) = server_chan::<IncomingPacket>();
+    let (peering_tx, peering_rx) = server_chan::<IncomingPacket>();
+    let incoming_senders = IncomingPacketSenders {
+        discovery_tx,
+        peering_tx,
+    };
+
+    // Event channel to publish events to the user.
+    let (event_tx, event_rx) = event::event_chan();
+
+    // Initialize the server managing the UDP socket I/O.
+    let server_config = ServerConfig::new(&config);
+    let (server, server_tx) = Server::new(server_config, local.clone(), incoming_senders);
+    server.init(&mut task_mngr).await;
+
+    // Create a request manager that creates and keeps track of outgoing requests.
+    let request_mngr = RequestManager::new(version, network_id, config.bind_addr());
+
+    // Create the discovery manager handling the discovery request/response protocol.
+    let discovery_config = DiscoveryManagerConfig::new(&config, version, network_id);
+    let discovery_socket = ServerSocket::new(discovery_rx, server_tx.clone());
+
+    let discovery_mngr = DiscoveryManager::new(
+        discovery_config,
+        local.clone(),
+        discovery_socket,
+        request_mngr.clone(),
+        peerstore.clone(),
+        master_peers.clone(),
+        active_peers.clone(),
+        replacements.clone(),
+        event_tx.clone(),
+    );
+    discovery_mngr.init(&mut task_mngr).await;
+
+    // Create neighborhoods and neighbor candidate filter.
+    let inbound_nbh = InboundNeighborhood::new();
+    let outbound_nbh = OutboundNeighborhood::new();
+    let nb_filter = NeighborFilter::new(local.peer_id(), neighbor_validator);
+
+    // Create the autopeering manager handling the peering request/response protocol.
+    let peering_socket = ServerSocket::new(peering_rx, server_tx.clone());
+
+    let peering_mngr = PeeringManager::new(
+        local.clone(),
+        peering_socket,
+        request_mngr.clone(),
+        active_peers.clone(),
+        event_tx.clone(),
+        inbound_nbh.clone(),
+        outbound_nbh.clone(),
+        nb_filter.clone(),
+    );
+    task_mngr.run(peering_mngr);
+
+    // TODO: remove this when sure that all open requests are garbage collected.
```

Comment:
Reminder

Reply:
I want to wait until more people are using it and providing bug reports, then address it in a patch version. But thanks for reminding me. :+1: